unified-runtime/source/adapters/level_zero/v2/usm.hpp

```diff
@@ -20,6 +20,34 @@
 
 struct UsmPool;
 
+struct AllocationStats {
+public:
+  enum UpdateType {
+    INCREASE,
+    DECREASE,
+  };
+
+  void update(UpdateType Type, size_t Size) {
+    if (Type == INCREASE) {
+      AllocatedMemorySize += Size;
+      size_t Current = AllocatedMemorySize.load();
```

Comment:
you might change these loads/stores to relaxes (std::memory_order_relaxed) - we don't really need sequential here and relaxed atomic would be faster.